src/main/java/org/alliancegenome/curation_api/model/entities/Annotation.java

```diff
@@ -0,0 +1,107 @@
+import org.hibernate.search.engine.backend.types.Searchable;
+import org.hibernate.search.engine.backend.types.Sortable;
+import org.hibernate.search.mapper.pojo.automaticindexing.ReindexOnUpdate;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.FullTextField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.IndexedEmbedded;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.IndexingDependency;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.KeywordField;
+
+import com.fasterxml.jackson.annotation.JsonView;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Audited
+@Entity
+@Data
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
+@Inheritance(strategy = InheritanceType.JOINED)
+@Schema(name = "association", description = "Annotation class representing a disease annotation")
+@AGRCurationSchemaVersion(min = "1.8.0", max = LinkMLSchemaConstants.LATEST_RELEASE, dependencies = { AuditedObject.class })
+
+@Table(indexes = { 
+	@Index(name = "annotation_createdby_index", columnList = "createdBy_id"), 
+	@Index(name = "annotation_updatedby_index", columnList = "updatedBy_id"),
+	@Index(name = "annotation_singlereference_index", columnList = "singleReference_curie"),
+	@Index(name = "annotation_dataprovider_index", columnList = "dataProvider_id"),
+})
+
+public class Annotation extends GeneratedAuditedObject {
+
+	@FullTextField(analyzer = "autocompleteAnalyzer", searchAnalyzer = "autocompleteSearchAnalyzer")
+	@KeywordField(name = "uniqueId_keyword", aggregable = Aggregable.YES, sortable = Sortable.YES, searchable = Searchable.YES, normalizer = "sortNormalizer")
+	@Column(length = 2000)
+	@JsonView({ View.FieldsOnly.class })
+	@EqualsAndHashCode.Include
+	protected String uniqueId;
+
+	@FullTextField(analyzer = "autocompleteAnalyzer", searchAnalyzer = "autocompleteSearchAnalyzer")
+	@KeywordField(name = "curie_keyword", aggregable = Aggregable.YES, sortable = Sortable.YES, searchable = Searchable.YES, normalizer = "sortNormalizer")
+	@JsonView({ View.FieldsOnly.class })
+	@EqualsAndHashCode.Include
+	protected String curie;
+
+	@FullTextField(analyzer = "autocompleteAnalyzer", searchAnalyzer = "autocompleteSearchAnalyzer")
+	@KeywordField(name = "modEntityId_keyword", aggregable = Aggregable.YES, sortable = Sortable.YES, searchable = Searchable.YES, normalizer = "sortNormalizer")
+	@Column(unique = true)
+	@JsonView({ View.FieldsOnly.class })
+	@EqualsAndHashCode.Include
+	private String modEntityId;
+	
+	@FullTextField(analyzer = "autocompleteAnalyzer", searchAnalyzer = "autocompleteSearchAnalyzer")
+	@KeywordField(name = "modInternalId_keyword", aggregable = Aggregable.YES, sortable = Sortable.YES, searchable = Searchable.YES, normalizer = "sortNormalizer")
+	@Column(unique = true)
+	@JsonView({ View.FieldsOnly.class })
+	@EqualsAndHashCode.Include
+	private String modInternalId;
```

Comment:
If these are going to be looked up by curie or any of these ids you might want indexes for them also

Reply:
Done